src/corehost/cli/fxr/fx_muxer.cpp

```diff
@@ -601,6 +607,16 @@ namespace
         return StatusCode::Success;
     }
 
+    bool should_check_framework_compatibility(const host_context_t *existing_context)
+    {
+        // Validate for framework-dependent (i.e. non-empty frameworks)
+        if (!existing_context->fx_versions_by_name.empty())
+            return true;
+
+        // Validate for non-3.0
```

Comment:
We should not do a version check (sorry for confusion since we discussed it like that) - the code path will only be hit for 3.0 and up (so 2.* is not affected anyway) and higher versions should continue to work this way until we implement a better fix which will either add a version check or handle this differently.